src/Core/WinSWCore/LogAppenders.cs

```diff
@@ -299,16 +309,189 @@ private void CopyStreamWithRotation(Stream data, string ext)
+
+                                // even if the log rotation fails, create a new one, or else
+                                // we'll infinitely try to rotate.
+                                w = new FileStream(logFile, FileMode.Create);
+                                sz = new FileInfo(logFile).Length;
+                            }
+                        }
+                        catch (Exception e)
+                        {
+                            EventLogger.LogEvent(string.Format("Failed to roll size time log: {0}", e.Message));
+                        }
+                    }
+                    w.Flush();
+                }
+            }
+            data.Close();
+            w.Close();
+        }
+
+        private double SetupRollTimer()
+        {
+            var nowTime = DateTime.Now.ToUniversalTime();
+            var scheduledTime = new DateTime(nowTime.Year, nowTime.Month, nowTime.Day, AutoRollAtTime.Value.Hours,
+                AutoRollAtTime.Value.Minutes, AutoRollAtTime.Value.Seconds, 0).ToUniversalTime(); //Specify your time HH,MM,SS
+            if (nowTime > scheduledTime)
+                scheduledTime = scheduledTime.AddDays(1);
+
+            double tickTime = (double) (scheduledTime - DateTime.Now.ToUniversalTime()).TotalMilliseconds;
+            return tickTime;
+        }
+
+        private int GetNextFileNumber(string ext, string baseDirectory, string baseFileName)
+        {
+            var nextFileNumber = 0;
+            var files = Directory.GetFiles(baseDirectory, String.Format("{0}.{1}.#*{2}", baseFileName, DateTime.UtcNow.ToString(FilePattern), ext));
+            if (files.Length == 0)
+            {
+                nextFileNumber = 1;
+            }
+            else
+            {
+                foreach (var f in files)
+                {
+                    try
+                    {
+                        var filenameOnly = Path.GetFileNameWithoutExtension(f);
+                        var lastNumberAsString = filenameOnly.Substring(filenameOnly.Length - 4, 4);
+                        int lastNumber = 0;
+                        if (int.TryParse(lastNumberAsString, out lastNumber))
+                        {
+                            if (lastNumber > nextFileNumber)
+                                nextFileNumber = lastNumber;
+                        }
+                        else
+                            throw new IOException(string.Format("File {0} does not follow the pattern provided",f));
+                    }
+                    catch (Exception e)
+                    {
+                        throw new IOException(string.Format("Failed to process file {0} due to error {1}",f, e.Message), e);
+                    }
+                }
+                if (nextFileNumber == 0) throw new IOException("Cannot roll the file because matching pattern not found");
+                nextFileNumber++;
+            }
+            return nextFileNumber;

```

Comment:
There is some copy-paste in this appender, but OK for now

Reply:
That is fine I will use my fork version until you merge it.

Reply:
Hi let me know when you planning to merge this into master.

Reply:
My optimistic plan - this Saturday